lib/src/rewrite.rs

```diff
@@ -130,6 +130,22 @@ pub fn rebase_commit_with_options(
     new_parents: &[Commit],
     options: &RebaseOptions,
 ) -> Result<RebasedCommit, TreeMergeError> {
+    // If specified, don't create commit where one parent is an ancestor of another.
+    let mut new_parents = new_parents.iter().cloned().collect_vec();
+    if options.simplify_ancestor_merge {
+        let new_parent_ids = new_parents
+            .iter()
+            .map(|commit| commit.id())
+            .cloned()
+            .collect_vec();
```

Comment:
nit: `.cloned().collect_vec()` can be removed. The iterator can be passed to  `.heads()`.